docs/content/docs/deep/componets.md

```diff
@@ -0,0 +1,113 @@
+# Components
+
+Fluid Framework powered applications have several ways to modularize their code and data into separate components.
+
+## Containers
+
+We expect most Fluid applications to to operate on a single container at a time, ans store all their persisted content in that container.
+It is however possible to use multiple containers, and some use-cases may require it.
+Specifically data which require different permissions, or to be stored in different locations, has to be split into separate containers.
+
+When splitting documents/data across multiple containers, it is up to the application to decide when to delete containers which are no longer needed.
+
+Each container is independent synchronization wise and there is no way to do anything atomically or even strongly ordered across multiple containers.
+
+Each container gets its own service client and thus separate op stream, storage and permissions.
+
+If desired, an application can open a second copy of a container as a mechanism to have a separate "branch" (independent view which can have edits speculatively applied).
+This is currently not well optimized (for performance, memory use or ergonomics) and the functionality of this is rather limited (its impractical to use it to preview changes, and keep rebasing them to stay up to date before committing them for example).
+
+An application using multiple containers can (though for bundle size generally shouldn't) use different versions or copies of the framework and its runtime for different containers.
+
+## Data Objects and Distributed Data Structures (DDSes)
+
+Each container gets a `ContainerSchema`, which defines a tree (currently implemented using shared directory) of statically defined DDSes and DataObjects.
+The container also can contain dynamically allocated DDSes and DataObjects which can be referenced from others via an `IFluidHandle`, and garbage collected when unreferenced.
+
+When configuring the container, the application specifies which DDSes and DataObjects implementations the container supports, which implicitly selects the concrete implementation (which package version and which copy of the package if there are multiple) the container will use for the DDses and DataObjects.
+
+Smaller libraries/components which the applications uses to work with data within the container should define version ranges of the Fluid client libraries they are compatible with, and the application should ensure all code using the same container is using the same copies of the Fluid client packages.
```

Comment:
I'd remove the word "smaller". Size is not a differentiator here.
Any sizable project would consist mostly out of libraries (and some of them are huge - Loop container is probably 10x smaller code-wise that all the components combined; Word team people-wise is substantially smaller than all shared teams contributing code to Word binary).